File: backend/WarehouseOps.Domain/Incident.cs
namespace WarehouseOps.Domain;

public class Incident : BaseEntity
{
    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public IncidentSeverity Severity { get; set; } = IncidentSeverity.Medium;

    public IncidentRelatedEntityType RelatedEntityType { get; set; } = IncidentRelatedEntityType.General;

    public string RelatedEntityId { get; set; } = string.Empty;

    public IncidentStatus Status { get; set; }

    public string ResolutionNotes { get; set; } = string.Empty;

    public DateTime? ClosedAt { get; set; }
}

File: frontend/warehouseops-client/src/types/incident.ts
export type Incident = {
  id: string;
  title: string;
  description: string;
  severity: string;
  relatedEntityType: string;
  relatedEntityId: string;
  status: string;
  resolutionNotes: string;
  closedAt: string | null;
  createdAt: string;
  updatedAt: string | null;
};

export type IncidentFilters = {
  status?: string;
};

export type CreateIncidentRequest = {
  title: string;
  description: string;
  severity: string;
  relatedEntityType: string;
  relatedEntityId: string;
};

export type UpdateIncidentStatusRequest = {
  status: string;
};

export type ResolveIncidentRequest = {
  resolutionNotes: string;
};
